admin/admin-functions.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * The admin-specific functionality of the plugin.
+ *
+ * @link       https://nataliesmith.dev
+ * @since      1.0.0
+ * @package    Poetwp
+ * @subpackage Poetwp/admin
+ */
+
+/**
+ * Add custom meta boxes to the poem post type
+ */
+function poetwp_add_meta_boxes() {
+	add_meta_box( 'poem_details', 'Poem Details', 'poetwp_poem_details_callback', 'poem', 'normal', 'default' );
+}
+// Hook the meta box function to the 'add_meta_boxes' action.
+add_action( 'add_meta_boxes', 'poetwp_add_meta_boxes' );
+
+/**
+ * Callback function to display the contents of the meta box
+ *
+ * @param WP_Post $post The current post object.
+ */
+function poetwp_poem_details_callback( $post ) {
```

Comment:
```suggestion
 * @param WP_Post $post The current post object.
 */
function poetwp_poem_details_callback( WP_Post $post ) {
```

Since this is a new plugin, and you know what the argument type should be, don't be afraid to add the type declaration in to the parameter. 

You can then also add a `declare(strict_types=1)` as the first statement in the file as well.

There is a whole argument about the use of strong typing in plugins (see https://peterwilson.cc/type-declarations-with-wordpress-filters/), but for me, the benefits of QA tooling + tests outweigh the potential cost of having something slightly broken (but it's easy enough to fix).